velox/substrait/tests/VeloxSubstraitRoundTripTest.cpp

```diff
@@ -68,6 +68,32 @@ class VeloxSubstraitRoundTripTest : public OperatorTestBase {
     // Convert Substrait Plan to the same Velox Plan.
     auto samePlan = substraitConverter_->toVeloxPlan(substraitPlan);
 
+    // Test projection.
```

Comment:
This logic is very specific to one kind of plans and should not be needed. It should be sufficient to have round-trip test for plans that are expected to fail and plans that are expected to succeed.

- run original plan; record failure or success + results.
- convert to Substrait and back to get a "new" plan.
- run the new plan and verify results match the original plan; either both plans fail or both succeed and return same results.

Once we have that, we can test cast using the following queries:

SELECT cast('abc' as bigint) ...

SELECT try_cast('abc' as bigint) ...

The first plan fails, the second succeeds. If we fail to preserve the null-on-failure flag during conversion the test would fail.

Reply:
Thanks for the comment!
Just removed the specific check for cast expr.
Not sure I correctly catch your point. In the new commit, we added two extra tests for casting VARCHAR to int64. And also made the assert function support checking expected failure.